From 16e553d51ccb484c20f37814bb44696a8f14d29a Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Fri, 19 Feb 2021 01:03:23 -0800
Subject: [PATCH 2/2] SKIP_INSTALL=NO for framework

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 Configurations/ConfigFramework.xcconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Configurations/ConfigFramework.xcconfig b/Configurations/ConfigFramework.xcconfig
index e3db1abd..03516805 100644
--- a/Configurations/ConfigFramework.xcconfig
+++ b/Configurations/ConfigFramework.xcconfig
@@ -7,7 +7,7 @@ WRAPPER_EXTENSION = framework
 FRAMEWORK_VERSION = A
 INFOPLIST_FILE = Sparkle/Sparkle-Info.plist
 GCC_PREPROCESSOR_DEFINITIONS = $(inherited) BUILDING_SPARKLE=1
-SKIP_INSTALL = YES
+SKIP_INSTALL = NO
 DEFINES_MODULE = YES
 PRODUCT_BUNDLE_IDENTIFIER = ${SPARKLE_BUNDLE_IDENTIFIER}
 
-- 
2.24.3 (Apple Git-128)

