diff --git a/src/c11/time.h b/src/c11/time.h
index 53d587c3eb6..13daa51dcde 100644
--- a/src/c11/time.h
+++ b/src/c11/time.h
@@ -37,9 +37,13 @@ struct timespec
 
 /*-------------------------- functions --------------------------*/
 
-#ifndef HAVE_TIMESPEC_GET
+#if !defined(HAVE_TIMESPEC_GET) || defined(__APPLE__)
 /*-------------------- 7.25.7 Time functions --------------------*/
 // 7.25.6.1
+#ifdef __APPLE__
+#include <time.h>
+#define timespec_get(ts, b) mesa_timespec_get(ts, b)
+#endif
 int
 timespec_get(struct timespec *ts, int base);
 #endif
