From 83a8ebf44cd23403cf9384a77ab3f72cca083c7c Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Tue, 5 Jan 2021 14:45:35 -0800
Subject: [PATCH 1/3] Enable otvalid

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 modules.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.cfg b/modules.cfg
index aded92e5e..9ba26c656 100644
--- a/modules.cfg
+++ b/modules.cfg
@@ -136,7 +136,7 @@ AUX_MODULES += bzip2
 
 # OpenType table validation.  Needs `ftotval.c' below.
 #
-# AUX_MODULES += otvalid
+AUX_MODULES += otvalid
 
 # Auxiliary PostScript driver component to share common code.
 #
-- 
2.32.0 (Apple Git-132)

