diff --git a/meson.build b/meson.build
index ab0fb13..c589db2 100644
--- a/meson.build
+++ b/meson.build
@@ -239,8 +239,8 @@ endforeach
 conf.set_quoted('FC_DEFAULT_FONTS', escaped_xml_path)
 fonts_conf.set('FC_DEFAULT_FONTS', xml_path)
 
-fc_templatedir = join_paths(prefix, get_option('datadir'), 'fontconfig/conf.avail')
-fc_baseconfigdir = join_paths(prefix, get_option('sysconfdir'), 'fonts')
+fc_baseconfigdir = join_paths(prefix, get_option('sysconfdir'), 'X11/fontconfig')
+fc_templatedir = join_paths(fc_baseconfigdir, 'conf.avail')
 fc_configdir = join_paths(fc_baseconfigdir, 'conf.d')
 fc_xmldir = join_paths(prefix, get_option('datadir'), 'xml/fontconfig')
 
