diff --git a/freeglut/src/freeglut_main.c b/freeglut/src/freeglut_main.c
index e54094a..58dfbb6 100644
--- freeglut/src/freeglut_main.c
+++ freeglut/src/freeglut_main.c
@@ -339,15 +339,9 @@ unsigned long fgSystemTime(void) {
     return timeGetTime();
 #    endif
 #else
-#   ifdef CLOCK_MONOTONIC
-    struct timespec now;
-    clock_gettime(CLOCK_MONOTONIC, &now);
-    return now.tv_nsec/1000000 + now.tv_sec*1000;
-#   elif HAVE_GETTIMEOFDAY
     struct timeval now;
     gettimeofday( &now, NULL );
     return now.tv_usec/1000 + now.tv_sec*1000;
-#   endif
 #endif
 }
   
