diff -Naurp freeglut-2.8.0.orig/configure.ac freeglut-2.8.0/configure.ac
--- freeglut-2.8.0.orig/configure.ac	2011-09-27 18:37:26.000000000 -0700
+++ freeglut-2.8.0/configure.ac	2012-03-16 22:23:33.000000000 -0700
@@ -28,6 +28,10 @@ if test "x$no_x" = xyes; then
   VERSION_INFO=
   EXPORT_FLAGS="-DFREEGLUT_EXPORTS"
 else
+  /* Append the include and lib path to CFLAGS and LDFLAGS */
+  CFLAGS="$CFLAGS $X_CFLAGS"
+  LDFLAGS="$LDFLAGS $X_LIBS"
+
   GL_LIBS="-lGL -lXext -lX11"
   AC_CHECK_LIB([Xxf86vm], [XF86VidModeSwitchToMode])
   AC_CHECK_LIB([Xrandr], [XRRQueryExtension])
